crates/scope/src/pass.rs

```diff
@@ -164,4 +194,20 @@ impl<'alloc> Pass<'alloc> for ScopePass<'alloc> {
     fn leave_function_body(&mut self, _ast: &'alloc FunctionBody<'alloc>) {
         self.builder.after_function_body();
     }
+
+    /// Arrow function with expression body.
+    /// Use the expression as the node for function body.
+    fn enter_enum_arrow_expression_body_variant_expression(
+        &mut self,
+        ast: &'alloc arena::Box<'alloc, Expression<'alloc>>,
+    ) {
+        self.builder.before_function_body(ast.deref());
```

Comment:
This `.deref()` method is almost never called directly. Here I think it is better to specify the desired type like this:

```rust
let expr: &Expression = ast;
self.builder.before_function_body(expr);
```